synapse/push/bulk_push_rule_evaluator.py

```diff
@@ -429,17 +434,22 @@ async def _action_for_event_by_user(
                     if not isinstance(display_name, str):
                         display_name = None
 
-            if count_as_unread:
-                # Add an element for the current user if the event needs to be marked as
-                # unread, so that add_push_actions_to_staging iterates over it.
-                # If the event shouldn't be marked as unread but should notify the
-                # current user, it'll be added to the dict later.
-                actions_by_user[uid] = []
-
             actions = evaluator.run(rules, uid, display_name)
-            if "notify" in actions:
-                # Push rules say we should notify the user of this event
-                actions_by_user[uid] = actions
+
+            # check whether unread counts are enabled for this user
+            unread_enabled = await self.store.get_feature_enabled(uid, "msc2654")
+            if not unread_enabled:
+                unread_enabled = self.hs.config.experimental.msc2654_enabled
```

Comment:
This will add a database query per user in the room on a very hot path. I think keeping this as a per-server config makes more sense.

Reply:
No problem - one question about this though: this is the second PR with code pertaining to enabling msc2654 per-user, and there's code from the first PR that would need to be deleted. Would it make sense to delete that code in this PR, or should I open a third PR to delete the code that has already been merged (it's not much, just things like a field in an enum, little pieces of the docs, part of a test, etc)?

Reply:
> I open a third PR to delete the code that has already been merged (it's not much, just things like a field in an enum, little pieces of the docs, part of a test, etc)?

I'd probably just use a separate PR, but that's me...